jetty-util/src/main/java/org/eclipse/jetty/util/thread/QueuedThreadPool.java

```diff
@@ -425,16 +424,34 @@ public int getMinThreads()
      * @return number of reserved threads or or -1 for heuristically determined
      * @see #setReservedThreads
      */
-    @ManagedAttribute("the number of reserved threads in the pool")
+    @ManagedAttribute("number of configured reserved threads")
     public int getReservedThreads()
     {
-        if (isStarted())
+        return _reservedThreads;
+    }
+
+    @ManagedAttribute("maximum number of reserved threads")
+    public int getMaxReservedThreads()
+    {
+        TryExecutor tryExecutor = _tryExecutor;
+        if (tryExecutor instanceof ReservedThreadExecutor)
         {
-            ReservedThreadExecutor reservedThreadExecutor = getBean(ReservedThreadExecutor.class);
-            if (reservedThreadExecutor != null)
-                return reservedThreadExecutor.getCapacity();
+            ReservedThreadExecutor reservedThreadExecutor = (ReservedThreadExecutor)tryExecutor;
+            return reservedThreadExecutor.getCapacity();
         }
-        return _reservedThreads;
+        return 0;
+    }
+
+    @ManagedAttribute("number of available reserved threads")
+    public int getAvailableReservedThreads()
```

Comment:
I don't like this name.  There is no such thing as an unavailable reserved thread.